fintual-oss/paraffin-back
# feat: add sign in with GitHub oauth

# Base PR
- PR based on main

# Context
- With this PR, it is possible to login with github with omniauth gem

# Changelog
- Add new gem: omniauth-github
- Add a new oauth2 provider (github)
- Add github sign in button to devise links view
- Add handle_auth method to `OmniauthCallbacksController` to handle multiple providers 
- Add new secrets to devise configuration 


# QA
- Create .env file with the credentials
- please check if you have a public email in your github account. Otherwise, it is not possible to use this registration method
- Run: docker compose build
- Run: docker compose run web rails db:reset
- Run: docker compose run web rails db:migrate
- [FRONT] On the main branch go to http://localhost:3001/users/sign_in
